lib/denylist/pg.go

```diff
@@ -0,0 +1,79 @@
+package denylist
+
+import (
+	"database/sql"
+	"sync"
+
+	_ "github.com/lib/pq"
+)
+
+type Syncer struct {
+	Persistent bool
+	Handle     *sql.DB
+}
+
+func NewSyncer(persistenceURL string) (*Syncer, error) {
+	if persistenceURL == "" {
+		return &Syncer{
+			Persistent: false,
+		}, nil
+	}
+
+	db, err := sql.Open("postgres", persistenceURL)
+	if err != nil {
+		return nil, err
+	}
+	return &Syncer{
+		Persistent: true,
+		Handle:     db,
+	}, nil
+}
+
+func (syncer *Syncer) LoadDenylist() (*sync.Map, error) {
+	if syncer.Persistent {
+		_, err := syncer.Handle.Exec("CREATE TABLE IF NOT EXISTS otr_denylist (entry VARCHAR(255) UNIQUE);")
```

Comment:
what database are we using here? this didn't occur to me, but we're not in any customer context. is this going to be a problem?

Reply:
The (physical)instance that I'm planning to use is the garden PG, in a different logical DB (something like `oplogtoredis`) -- Given that the load for this probably isn't much at all and most calls are O(1), but TBD John's review in the chart PR

Reply:
the chart PR is now merged - we should be good w that decision here